Unable to install beta and prod variants simultaneously

**Summary:** 

After #1725 got fixed in #1726, beta and prod variants should coexist on the same device. Recent changes have again started build failures when trying to install beta over prod. 

**Steps to reproduce:** 

Install prod variant. Now try installing beta variant. 

**Add System logs:**

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:installBetaDebug'.
> com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: INSTALL_FAILED_CONFLICTING_PROVIDER: Package couldn't be installed in /data/app/fr.free.nrw.commons.beta-1: Can't install because provider name fr.free.nrw.commons.explore.recentsearches.contentprovider (in package fr.free.nrw.commons.beta) is already used by fr.free.nrw.commons

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 10s
35 actionable tasks: 3 executed, 32 up-to-date

```

 **Commons app version:** 

`master`

**Would you like to work on the issue?**

Yes

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -149,6 +149,7 @@ android {
             buildConfigField "String", "CONTRIBUTION_AUTHORITY", "\"fr.free.nrw.commons.contributions.contentprovider\""
             buildConfigField "String", "MODIFICATION_AUTHORITY", "\"fr.free.nrw.commons.modifications.contentprovider\""
             buildConfigField "String", "CATEGORY_AUTHORITY", "\"fr.free.nrw.commons.categories.contentprovider\""
+            buildConfigField "String", "RECENT_SEARCH_AUTHORITY", "\"fr.free.nrw.commons.explore.recentsearches.contentprovider\""
 
             dimension 'tier'
         }
@@ -175,6 +176,7 @@ android {
             buildConfigField "String", "CONTRIBUTION_AUTHORITY", "\"fr.free.nrw.commons.beta.contributions.contentprovider\""
             buildConfigField "String", "MODIFICATION_AUTHORITY", "\"fr.free.nrw.commons.beta.modifications.contentprovider\""
             buildConfigField "String", "CATEGORY_AUTHORITY", "\"fr.free.nrw.commons.beta.categories.contentprovider\""
+            buildConfigField "String", "RECENT_SEARCH_AUTHORITY", "\"fr.free.nrw.commons.beta.explore.recentsearches.contentprovider\""
 
             dimension 'tier'
         }

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -190,7 +190,7 @@
 
         <provider
             android:name=".explore.recentsearches.RecentSearchesContentProvider"
-            android:authorities="fr.free.nrw.commons.explore.recentsearches.contentprovider"
+            android:authorities="${applicationId}.explore.recentsearches.contentprovider"
             android:exported="false"
             android:label="@string/provider_searches"
             android:syncable="false" />

--- a/app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java
+++ b/app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java
@@ -11,7 +11,6 @@ import javax.inject.Singleton;
 
 import dagger.Module;
 import dagger.Provides;
-
 import fr.free.nrw.commons.BuildConfig;
 import fr.free.nrw.commons.auth.AccountUtil;
 import fr.free.nrw.commons.auth.SessionManager;
@@ -24,7 +23,6 @@ import fr.free.nrw.commons.wikidata.WikidataEditListener;
 import fr.free.nrw.commons.wikidata.WikidataEditListenerImpl;
 
 import static android.content.Context.MODE_PRIVATE;
-import static fr.free.nrw.commons.explore.recentsearches.RecentSearchesContentProvider.RECENT_SEARCH_AUTHORITY;
 
 @Module
 @SuppressWarnings({"WeakerAccess", "unused"})
@@ -60,7 +58,7 @@ public class CommonsApplicationModule {
     @Provides
     @Named("recentsearch")
     public ContentProviderClient provideRecentSearchContentProviderClient(Context context) {
-        return context.getContentResolver().acquireContentProviderClient(RECENT_SEARCH_AUTHORITY);
+        return context.getContentResolver().acquireContentProviderClient(BuildConfig.RECENT_SEARCH_AUTHORITY);
     }
 
     @Provides

--- a/app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesContentProvider.java
+++ b/app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesContentProvider.java
@@ -11,6 +11,7 @@ import android.text.TextUtils;
 
 import javax.inject.Inject;
 
+import fr.free.nrw.commons.BuildConfig;
 import fr.free.nrw.commons.contributions.ContributionDao;
 import fr.free.nrw.commons.data.DBOpenHelper;
 import fr.free.nrw.commons.di.CommonsDaggerContentProvider;
@@ -28,17 +29,16 @@ import static fr.free.nrw.commons.explore.recentsearches.RecentSearchesDao.Table
  **/
 public class RecentSearchesContentProvider extends CommonsDaggerContentProvider {
 
-    public static final String RECENT_SEARCH_AUTHORITY = "fr.free.nrw.commons.explore.recentsearches.contentprovider";
     // For URI matcher
     private static final int RECENT_SEARCHES = 1;
     private static final int RECENT_SEARCHES_ID = 2;
     private static final String BASE_PATH = "recent_searches";
-    public static final Uri BASE_URI = Uri.parse("content://" + RECENT_SEARCH_AUTHORITY + "/" + BASE_PATH);
+    public static final Uri BASE_URI = Uri.parse("content://" + BuildConfig.RECENT_SEARCH_AUTHORITY + "/" + BASE_PATH);
     private static final UriMatcher uriMatcher = new UriMatcher(NO_MATCH);
 
     static {
-        uriMatcher.addURI(RECENT_SEARCH_AUTHORITY, BASE_PATH, RECENT_SEARCHES);
-        uriMatcher.addURI(RECENT_SEARCH_AUTHORITY, BASE_PATH + "/#", RECENT_SEARCHES_ID);
+        uriMatcher.addURI(BuildConfig.RECENT_SEARCH_AUTHORITY, BASE_PATH, RECENT_SEARCHES);
+        uriMatcher.addURI(BuildConfig.RECENT_SEARCH_AUTHORITY, BASE_PATH + "/#", RECENT_SEARCHES_ID);
     }
 
     public static Uri uriForId(int id) {

